src/components/Session/SessionFilter.vue

```diff
@@ -105,6 +111,36 @@ export default defineComponent({
       router.back()
     }
 
+    const checkMySchedule = () => {
+      const events: any[] = [];
+      favoriteSessions.value.forEach(function(SessionId) {
+        const session = computed(() => getSessionById(SessionId));
+        events.push({"startTime": new Date(session.value.start), "endTime": new Date(session.value.end)});
+      });
+      
+      var isConflict = false;
+      for (let i = 0; i < events.length - 1; i++) {
+        for (let j = i + 1; j < events.length; j++) {
+          const eventA = events[i];
+          const eventB = events[j];
+          if ( 
+            (eventA["endTime"] > eventB["startTime"] && eventA["startTime"] < eventB["endTime"]) || 
+            (eventB["endTime"] > eventA["startTime"] && eventB["startTime"] < eventA["endTime"]) 
+          ) {
+            isConflict = true;
+          }
+        }
+      }
```

Comment:
```suggestion
      comparisonLoop:
      for (let i = 0; i < events.length - 1; i++) {
        for (let j = i + 1; j < events.length; j++) {
          const eventA = events[i];
          const eventB = events[j];
          if ( 
            (eventA["endTime"] > eventB["startTime"] && eventA["startTime"] < eventB["endTime"]) || 
            (eventB["endTime"] > eventA["startTime"] && eventB["startTime"] < eventA["endTime"]) 
          ) {
            isConflict = true;
            break comparisonLoop;
          }
        }
      }
```

可以利用 labels 在發現衝突時提早離開迴圈，加快計算速度。
[StackOverflow 上的說明](https://stackoverflow.com/a/183197/11631322)

Reply:
原本想試試 Use label 的方法，但 ESLint 檢查會跑不過（error: Unexpected labeled statement & Unexpected label in break statement），因此就換其他方式了。

調整後結果，見 [9debe0d](https://github.com/COSCUP/2024/commit/9debe0daccc49b64c031c9113218654066dca9ac)

Reply:
我想到的一個做法，可以參考看看：
```ts
const checkMySchedule = () => {
  const events = favoriteSessions.value.map((sessionId) => {
    const session = getSessionById(sessionId)
    return { startTime: new Date(session.start), endTime: new Date(session.end) }
  })

  const isConflicting = () => {
    for (let i = 0; i < events.length - 1; i++) {
      for (let j = i + 1; j < events.length; j++) {
        const eventA = events[i]
        const eventB = events[j]
        if (
          (eventA.endTime > eventB.startTime && eventA.startTime < eventB.endTime) ||
          (eventB.endTime > eventA.startTime && eventB.startTime < eventA.endTime)
        ) {
          return true
        }
      }
    }
    return false
  }

  if (isConflicting()) {
    window.alert(t('session.schedule_conflicts'))
  } else {
    window.alert(t('session.schedule_no_conflicts'))
  }
}
```

1. 直接把 `favoriteSessions` map 成 `events`: 這樣就不用定義型別，TS 能自行從 `map` 推論
2. 把檢查用巢狀迴圈寫成函數: 避開需要使用標籤或旗幟的問題

Reply:
哇！code 真的乾淨許多，謝謝你。
我採用了你的做法，見 [cccb9a8](https://github.com/COSCUP/2024/commit/cccb9a83fc845360e439608e7e5319ff3cae868d)